### PR TITLE
EPMLSTRTAI 652

### PR DESCRIPTION
**Pull request by IvanovYura**
1. Added pagination for general stepSuggestions and byType in StepSuggestionController.java:
getStepsSuggestions() and getStepsSuggestionsByType() methods
2. Were added 2 overloaded methods to StepSuggestionService.java to get List:
getStepsSuggestion() and getStepsSuggestionByType()
3. Were implemented tests for new controller methods and for service

### CODE REVIEW

File: src/main/java/com/epam/test_generator/controllers/StepSuggestionController.java [hunk 1]
```diff
@@ -44,6 +44,54 @@
...
+                    paramType = "path"),
+            @ApiImplicitParam(
+                    name = "Authorization",
+                    value = "add here your token",
+                    paramType = "header",
+                    dataType = "string",
+                    required = true)
+    })
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_ENGINEER", "ROLE_TEST_LEAD", "ROLE_GUEST"})
+    @RequestMapping(value = "/stepSuggestions/{stepType}/page/{pageNumber}/size/{pageSize}",
```
Comment (hdghg) [inline src/main/java/com/epam/test_generator/controllers/StepSuggestionController.java]: stepType, pageNumber and pageSize are not resources, therefore should not be declared as PathVariables. Use RequestParam for these parameters. http://www.baeldung.com/rest-api-pagination-in-spring

Comment (hdghg) [reply src/main/java/com/epam/test_generator/controllers/StepSuggestionController.java]: Btw method getStepsSuggestionsByType follows wrong crud design pattern. I'd recommend you implement new one which takes stepType as request parameter, not path variable. Leave old method (endpoint) for backward compatibility and mark it as deprecated.

File: src/main/java/com/epam/test_generator/controllers/StepSuggestionController.java [hunk 2]
```diff
@@ -44,6 +44,54 @@
...
 
+    @ApiOperation(value = "Get page of step suggestions", nickname = "getStepsSuggestions")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = StepSuggestionDTO.class,
+                    responseContainer = "List")
+    })
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_ENGINEER", "ROLE_TEST_LEAD", "ROLE_GUEST"})
+    @RequestMapping(value = "/stepSuggestions/page/{pageNumber}/size/{pageSize}", method = RequestMethod.GET,
+            produces = "application/json")
+    @ApiImplicitParam(name = "Authorization", value = "add here your token", paramType = "header", dataType = "string", required = true)
```
Comment (hdghg) [inline src/main/java/com/epam/test_generator/controllers/StepSuggestionController.java]: And apply TAI codestyle

Comment (IvanovYura) [reply src/main/java/com/epam/test_generator/controllers/StepSuggestionController.java]: Currently I have a problem with applying TAI codestyle, will fix it sooner

**Pull request merged**